pkgs/by-name/en/ente-web/package.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchYarnDeps,
+  nodejs,
+  yarnConfigHook,
+  yarnBuildHook,
+  extraBuildEnv ? null,
+
+}:
+stdenv.mkDerivation (finalAttrs: rec {
+
+  pname = "ente-web";
+  version = "0.9.5";
+
+  src =
+    fetchFromGitHub {
+      owner = "ente-io";
+      repo = "ente";
+      sparseCheckout = [ "web" ];
+      rev = "photos-v${version}";
+      fetchSubmodules = true;
+      hash = "sha256-YJuhdMrgOQW4+LaxEvZNmFZDlFRBmPZot8oUdACdhhE=";
+    }
+    + "/web";
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${finalAttrs.src}/yarn.lock";
+    hash = "sha256-ZGZkpHZD2LoMIXzpQRAO4Fh9Jf4WxosgykKnn7I1+2g=";
+  };
+
+  nativeBuildInputs = [
+    yarnConfigHook
+    yarnBuildHook
+    nodejs
+  ];
+
+  env = lib.optionals (extraBuildEnv != null) extraBuildEnv;
+  # { NEXT_PUBLIC_ENTE_ENDPOINT = publicEndpoint; };
+  # env = lib.optionals (albumsEndpoint != null) { NEXT_PUBLIC_ENTE_ALBUMS_ENDPOINT = albumsEndpoint; };
```

Comment:
?

How about using only a single `publicEndpoint` argument to the package? It'd also be nice to document the default publicEndpoint using the `?` default argument syntax.

Reply:
I'm still working this out, that's why I marked the PR as WIP. It seems the build accepts more env vars and the build procces does different things depending on which of those are set. The defaults behaviour is when nothing is set, so I think it makes sense to make the whole env default to not being set.